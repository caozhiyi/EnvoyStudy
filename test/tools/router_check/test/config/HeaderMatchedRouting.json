{
  "virtual_hosts": [
    {
      "name": "local_service",
      "domains": ["*"],
      "routes": [
        {
          "prefix": "/",
          "cluster": "local_service_with_headers",
          "headers" : [
            {"name": "test_header", "value": "test"}
          ]
        },
        {
          "prefix": "/",
          "cluster": "local_service_with_multiple_headers",
          "headers" : [
            {"name": "test_header_multiple1", "value": "test1"},
            {"name": "test_header_multiple2", "value": "test2"}
          ]
        },
        {
          "prefix": "/",
          "cluster": "local_service_with_empty_headers",
          "headers" : [
            {"name": "test_header_presence"}
          ]
        },
        {
          "prefix": "/",
          "cluster": "local_service_with_header_pattern_set_regex",
          "headers" : [
            {"name": "test_header_pattern", "value": "^user=test-\\d+$", "regex": true}
          ]
        },
        {
          "prefix": "/",
          "cluster": "local_service_with_header_pattern_unset_regex",
          "headers" : [
            {"name": "test_header_pattern", "value": "^customer=test-\\d+$"}
          ]
        },
        {
          "prefix": "/",
          "cluster": "local_service_without_headers"
        }
      ]
    }
  ]
}
