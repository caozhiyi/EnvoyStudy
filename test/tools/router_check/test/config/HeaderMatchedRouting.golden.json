[
  {
    "test_name": "Test_1",
    "input": {
      ":authority": "www.lyft.com",
      ":path": "/"
    },
    "validate": {"cluster_name": "local_service_without_headers"}
  },
  {
    "test_name": "Test_2",
    "input": {
      ":authority": "www.lyft.com",
      ":path": "/",
      "additional_headers": [
        {
	  "field": "test_header",
	  "value": "test"
        }
      ]
    },
    "validate": {"cluster_name": "local_service_with_headers"}
  },
  {
    "test_name": "Test_3",
    "input": {
      ":authority": "www.lyft.com",
      ":path": "/",
      "additional_headers": [
        {
	  "field": "test_header_multiple1",
	  "value": "test1"
        },
        {
	  "field": "test_header_multiple2",
	  "value": "test2"
        }
      ]
    },
    "validate": {"cluster_name": "local_service_with_multiple_headers"}
  },
  {
    "test_name": "Test_4",
    "input": {
      ":authority": "www.lyft.com",
      ":path": "/",
      "additional_headers": [
        {
	  "field": "non_existent_header",
	  "value": "foo"
        }
      ]
    },
    "validate": {"cluster_name": "local_service_without_headers"}
  },
  {
    "test_name": "Test_5",
    "input": {
      ":authority": "www.lyft.com",
      ":path": "/",
      "additional_headers": [
        {
	  "field": "test_header_presence",
	  "value": "test"
        }
      ]
    },
    "validate": {"cluster_name": "local_service_with_empty_headers"}
  },
  {
    "test_name": "Test_6",
    "input": {
      ":authority": "www.lyft.com",
      ":path": "/",
      "additional_headers": [
        {
	  "field": "test_header_pattern",
	  "value": "user=test-1223"
        }
      ]
    },
    "validate": {"cluster_name": "local_service_with_header_pattern_set_regex"}
  },
  {
    "test_name": "Test_7",
    "input": {
      ":authority": "www.lyft.com",
      ":path": "/",
      "additional_headers": [
        {
          "field": "test_header_pattern",
          "value": "customer=test-1223"
        }
      ]
    },
    "validate": {"cluster_name": "local_service_without_headers"}
  }
]
